src/ui/components/OrderList/components/OrderListItem.tsx

```diff
@@ -0,0 +1,147 @@
+					<div className="flex flex-row items-center justify-between py-4 md:flex-col md:items-start md:gap-y-1">
+						<dt className="font-medium text-neutral-900">Order number</dt>
+						<dd className="text-neutral-600">{order.number}</dd>
+					</div>
+					<div className="flex flex-row items-center justify-between py-4 md:flex-col md:items-start md:gap-y-1">
+						<dt className="font-medium text-neutral-900">Date placed</dt>
+						<dd className="text-neutral-600">
+							<time dateTime={order.created}>{formatDate(new Date(order.created))}</time>
+						</dd>
+					</div>
+					<div className="flex flex-row items-center justify-between py-4 md:flex-col md:items-start md:gap-y-1">
+						<dt className="font-medium text-neutral-900">Payment status</dt>
+						<dd>
+							<PaymentStatus status={order.paymentStatus} />
+						</dd>
+					</div>
+				</dl>
+				{/* TODO: Reveal after implementing the order details page. */}
+				{/* <div className="flex flex-col md:col-span-1 md:flex-row md:items-center lg:col-span-2">
+					<Link
+						href="#"
+						className="flex items-center justify-center rounded border border-neutral-200 bg-white px-4 py-2 text-sm hover:bg-neutral-50 focus:bg-neutral-50 md:ml-auto"
+					>
+						View Order
+					</Link>
+				</div> */}
+			</div>
+
+			{order.lines.length > 0 && (
+				<>
+					<div className="md:border-x md:px-6">
+						<table className="w-full text-sm text-neutral-500">
+							<thead className="sr-only">
+								<tr>
+									<td>product</td>
+									<td className="max-md:hidden">quantity and unit price</td>
+									<td>price</td>
+								</tr>
+							</thead>
+							<tbody className="md:divide-y">
+								{order.lines.map((item) => {
+									if (item.variant) {
+										const product = item.variant?.product;
+
+										return (
+											<tr
+												className="[&>td:last-child]:text-end max-md:[&>td:not(:first-child):not(:last-child)]:hidden [&>td:not(:last-child)]:pr-6 md:[&>td:not(:last-child)]:pr-6 [&>td]:py-6"
+												key={product.id}
+											>
+												<td>
+													<div className="flex flex-row items-center">
+														{product.thumbnail && (
+															<div className="mr-3 aspect-square h-16 w-16 flex-shrink-0 overflow-hidden rounded-md border bg-neutral-50 md:mr-6 md:h-24 md:w-24">
+																<Image
+																	src={product.thumbnail.url}
+																	alt={product.thumbnail.alt ?? ""}
+																	width={200}
+																	height={200}
+																	className="h-full w-full object-contain object-center"
+																/>
+															</div>
+														)}
+														<div>
+															<Link
+																href={
+																	item.variant?.id
+																		? getHrefForVariant({
+																				productSlug: product.slug,
+																				variantId: item.variant.id,
+																		  })
+																		: `/products/${product.slug}`
```

Comment:
Below I've suggested to make the `variantId` in `getHrefForVariant` optional so this condition wouldn't be needed.